merlin/analysis/preprocess.py

```diff
@@ -134,3 +134,33 @@ def _preprocess_image(self, inputImage: np.ndarray) -> np.ndarray:
             filteredImage, deconFilterSize, self._deconSigma,
             self._deconIterations).astype(np.uint16)
         return deconvolvedImage
+
+
+class DeconvolutionPreprocessGuo(DeconvolutionPreprocess):
+
+    def __init__(self, dataSet, parameters=None, analysisName=None):
+        super().__init__(dataSet, parameters, analysisName)
+
+        # FIXME: This doesn't work as hoped because 'decon_iterations'
+        #        is added in the super-class. So we end up with 20
+        #        iterations instead of the 2 that we want to be the
+        #        default. And we can't do this before calling the
+        #        super() because then self.parameters won't exist.
+        #
```

Comment:
Could we check if 'decon_iterations' is in parameters instead of self.parameters? This could tell us whether the user already specified decon_iterations or whether it was just set to the standard default.

Reply:
Thanks for the suggestion, that seems to work.